streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1455,6 +1455,35 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
         }
     }
 
+  /**
+   * Zips two streams together with a specified function.
+   * Use the latest element of another stream if no element is available from another stream.
+   */
+  final def zipWithLatest[R1 <: R, E1 >: E, B, C](that: ZStream[R1, E1, B])(
+    f: (A, B) => C
+  ): ZStream[R1, E1, C] = {
+    var latestLeft  = null.asInstanceOf[A]
+    var latestRight = null.asInstanceOf[B]
+    val first = self.take(1).zip(that.take(1)).map { x =>
+      latestLeft = x._1
+      latestRight = x._2
+      f.tupled(x)
+    }
+    first ++ self
```

Comment:
Unfortunately this would execute the original stream twice. So for example, if the stream acquires a resource, it would be acquired twice, etc. Would be pretty surprising.

Reply:
I thought of using the same method of zipWith. Lately, I found mergeEither combinator is good enough for zipWithLatest. I've modified my implementation. I think it is a lot simpler.

Reply:
I wonder what is the correct behaviour of zipWithLatest. What should I return in the following situation?
![IMG_20190731_125743](https://user-images.githubusercontent.com/32609395/62185065-6fdf9980-b393-11e9-88cc-472ba9627e3e.jpg)
If it is (f5, s1), then my implementation is correct.